### PR TITLE
Rename workspace scope from designer to design

### PR DESCRIPTION
**Pull request by sonicyeti**
This PR covers the shift away from "designer" as a workspace scope name in favor of "design".
• Existing workspaces
• Imported workspaces from (legacy core & designer)
• Updated for export (from insomnia)

Fixes INS-533

### CODE REVIEW

File: packages/insomnia-app/app/common/migrate-from-designer.js [hunk 1]
```diff
@@ -214,7 +214,7 @@ export default async function migrateFromDesigner({
       // For each workspace coming from Designer, mark workspace.scope as 'designer'
```
Comment (develohpanda) [inline packages/insomnia-app/app/common/migrate-from-designer.js]: Need to update the comment here as well, or replace/unbind it altogether
```suggestion
      // For each workspace coming from Designer, mark workspace.scope as 'design'
```

File: packages/insomnia-app/app/models/workspace.js [hunk 2]
```diff
@@ -121,16 +121,21 @@ async function _migrateEnsureName(workspace: Workspace): Promise<Workspace> {
 function _migrateScope(workspace: Workspace): Workspace {
   if (
     workspace.scope === WorkspaceScopeKeys.designer ||
```
Comment (develohpanda) [inline packages/insomnia-app/app/models/workspace.js]: This first line needs to be removed - `WorkspaceScopeKeys.designer` no longer exists. TS will save us here in the future.

File: packages/insomnia-app/app/models/workspace.js [hunk 3]
```diff
@@ -121,16 +121,21 @@ async function _migrateEnsureName(workspace: Workspace): Promise<Workspace> {
...
+    workspace.scope === WorkspaceScopeKeys.design ||
     workspace.scope === WorkspaceScopeKeys.collection
   ) {
     return workspace;
   }
 
   // Translate the old value
-  type OldScopeTypes = 'spec' | 'debug' | null;
+  type OldScopeTypes = 'spec' | 'debug' | 'designer' | null;
```
Comment (develohpanda) [inline packages/insomnia-app/app/models/workspace.js:130]: 👍🏽

**Review by develohpanda — CHANGES_REQUESTED** [review]
Since there are changes directly related to it, it would be good to convert the hard-coded instances of `design` which you have found, to use the flow enum `WorkspaceScopeKeys.design`. It'll save us from having to work through the code-base again looking for things to replace. Whether you want to include that in this PR or raise a subsequent one is up to you, although I encourage doing it in this PR.

Other than that the changes look good, just found a couple of minor things that need cleanup

**Review by DMarby — APPROVED** [review]

**Review by DMarby — APPROVED** [review]

**Review by develohpanda — APPROVED** [review]

**Pull request merged**